test_remote_storage_upload_queue_retries is flaky

```
/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/allure_commons/_allure.py:221: in __call__
    return self._fixture_function(*args, **kwargs)
test_runner/fixtures/neon_fixtures.py:1126: in neon_env_builder
    yield builder
test_runner/fixtures/neon_fixtures.py:854: in __exit__
    self.env.pageserver.assert_no_errors()
test_runner/fixtures/neon_fixtures.py:1704: in assert_no_errors
    assert not errors
E   AssertionError: assert not ['2023-06-12T14:01:15.991913Z  WARN attach{tenant_id=0c63051cac6ce39d3250f0cf1520241f}:load_remote_timeline{timeline_id=15f41063eec5222a8d19d2fcbe33e481}:reconcile_with_remote: found future image layer 000000067F00000005000040000000000050-000000067F00000005000040030000000000__0000000001F27B88 on timeline 15f41063eec5222a8d19d2fcbe33e481 remote_consistent_lsn is 0/1F27948\n', '2023-06-12T14:01:15.992185Z  WARN attach{tenant_id=0c63051cac6ce39d3250f0cf1520241f}:load_remote_timeline{timeline_id=15f41063eec5222a8d19d2fcbe33e481}:reconcile_with_remote: found future image layer 000000067F00000005000040040100000000-000000067F00000005000040050000000010__0000000001F27B88 on timeline 15f41063eec5222a8d19d2fcbe33e481 remote_consistent_lsn is 0/1F27948\n', '2023-06-12T14:01:15.992374Z  WARN attach{tenant_id=0c63051cac6ce39d3250f0cf1520241f}:load_remote_timeline{timeline_id=15f41063eec5222a8d19d2fcbe33e481}:reconcile_with_remote: found future image layer 000000067F00000005000040000000000040-000000067F00000005000040000000000050__0000000001F27B88 on timeline 15f41063eec5222a8d19d2fcbe33e481 remote_consistent_lsn is 0/1F27948\n', '2023-06-12T14:01:15.992528Z  WARN attach{tenant_id=0c63051cac6ce39d3250f0cf1520241f}:load_remote_timeline{timeline_id=15f41063eec5222a8d19d2fcbe33e481}:reconcile_with_remote: found future image layer 000000067F00000005000040000000000020-000000067F00000005000040000000000030__0000000001F27B88 on timeline 15f41063eec5222a8d19d2fcbe33e481 remote_consistent_lsn is 0/1F27948\n', '2023-06-12T14:01:15.992776Z  WARN attach{tenant_id=0c63051cac6ce39d3250f0cf1520241f}:load_remote_timeline{timeline_id=15f41063eec5222a8d19d2fcbe33e481}:reconcile_with_remote: found future image layer 000000067F000000050000313C0100000000-000000067F00000005000040000000000010__0000000001F27B88 on timeline 15f41063eec5222a8d19d2fcbe33e481 remote_consistent_lsn is 0/1F27948\n', '2023-06-12T14:01:15.992803Z  WARN attach{tenant_id=0c63051cac6ce39d3250f0cf1520241f}:load_remote_timeline{timeline_id=15f41063eec5222a8d19d2fcbe33e481}:reconcile_with_remote: found future image layer 000000067F00000005000040050000000010-000000067F00000005000040050000000020__0000000001F27B88 on timeline 15f41063eec5222a8d19d2fcbe33e481 remote_consistent_lsn is 0/1F27948\n', ...]
```

https://neon-github-public-dev.s3.amazonaws.com/reports/pr-4468/5244173359/index.html#categories/300352a30621e1caca2466d84a6ae8a0/6df502bdd9d463b6/

--- a/pageserver/src/tenant/timeline.rs
+++ b/pageserver/src/tenant/timeline.rs
@@ -1774,7 +1774,7 @@ impl Timeline {
             match remote_layer_name {
                 LayerFileName::Image(imgfilename) => {
                     if imgfilename.lsn > up_to_date_disk_consistent_lsn {
-                        warn!(
+                        info!(
                         "found future image layer {} on timeline {} remote_consistent_lsn is {}",
                         imgfilename, self.timeline_id, up_to_date_disk_consistent_lsn
                     );
@@ -1799,7 +1799,7 @@ impl Timeline {
                     // is 102, then it might not have been fully flushed to disk
                     // before crash.
                     if deltafilename.lsn_range.end > up_to_date_disk_consistent_lsn + 1 {
-                        warn!(
+                        info!(
                             "found future delta layer {} on timeline {} remote_consistent_lsn is {}",
                             deltafilename, self.timeline_id, up_to_date_disk_consistent_lsn
                         );

--- a/pageserver/src/tenant/timeline.rs
+++ b/pageserver/src/tenant/timeline.rs
@@ -1600,7 +1600,7 @@ impl Timeline {
             if let Some(imgfilename) = ImageFileName::parse_str(&fname) {
                 // create an ImageLayer struct for each image file.
                 if imgfilename.lsn > disk_consistent_lsn {
-                    warn!(
+                    info!(
                         "found future image layer {} on timeline {} disk_consistent_lsn is {}",
                         imgfilename, self.timeline_id, disk_consistent_lsn
                     );
@@ -1632,7 +1632,7 @@ impl Timeline {
                 // is 102, then it might not have been fully flushed to disk
                 // before crash.
                 if deltafilename.lsn_range.end > disk_consistent_lsn + 1 {
-                    warn!(
+                    info!(
                         "found future delta layer {} on timeline {} disk_consistent_lsn is {}",
                         deltafilename, self.timeline_id, disk_consistent_lsn
                     );

--- a/test_runner/regress/test_gc_cutoff.py
+++ b/test_runner/regress/test_gc_cutoff.py
@@ -14,10 +14,6 @@ from fixtures.neon_fixtures import NeonEnvBuilder, PgBin
 def test_gc_cutoff(neon_env_builder: NeonEnvBuilder, pg_bin: PgBin):
     env = neon_env_builder.init_start()
 
-    # These warnings are expected, when the pageserver is restarted abruptly
-    env.pageserver.allowed_errors.append(".*found future image layer.*")
-    env.pageserver.allowed_errors.append(".*found future delta layer.*")
-
     pageserver_http = env.pageserver.http_client()
 
     # Use aggressive GC and checkpoint settings, so that we also exercise GC during the test

--- a/test_runner/regress/test_pageserver_restart.py
+++ b/test_runner/regress/test_pageserver_restart.py
@@ -72,10 +72,6 @@ def test_pageserver_restart(neon_env_builder: NeonEnvBuilder):
 def test_pageserver_chaos(neon_env_builder: NeonEnvBuilder):
     env = neon_env_builder.init_start()
 
-    # These warnings are expected, when the pageserver is restarted abruptly
-    env.pageserver.allowed_errors.append(".*found future image layer.*")
-    env.pageserver.allowed_errors.append(".*found future delta layer.*")
-
     # Use a tiny checkpoint distance, to create a lot of layers quickly.
     # That allows us to stress the compaction and layer flushing logic more.
     tenant, _ = env.neon_cli.create_tenant(

--- a/test_runner/regress/test_recovery.py
+++ b/test_runner/regress/test_recovery.py
@@ -15,10 +15,6 @@ def test_pageserver_recovery(neon_env_builder: NeonEnvBuilder):
     env = neon_env_builder.init_start()
     env.pageserver.is_testing_enabled_or_skip()
 
-    # These warnings are expected, when the pageserver is restarted abruptly
-    env.pageserver.allowed_errors.append(".*found future delta layer.*")
-    env.pageserver.allowed_errors.append(".*found future image layer.*")
-
     # Create a branch for us
     env.neon_cli.create_branch("test_pageserver_recovery", "main")
 

--- a/test_runner/regress/test_remote_storage.py
+++ b/test_runner/regress/test_remote_storage.py
@@ -348,9 +348,6 @@ def test_remote_storage_upload_queue_retries(
     # XXX: should vary this test to selectively fail just layer uploads, index uploads, deletions
     #      but how do we validate the result after restore?
 
-    # these are always possible when we do an immediate stop. perhaps something with compacting has changed since.
-    env.pageserver.allowed_errors.append(r".*found future (delta|image) layer.*")
-
     env.pageserver.stop(immediate=True)
     env.endpoints.stop_all()
 

